keys/keys.go

```diff
@@ -70,6 +70,42 @@ func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
 	return k
 }
 
+// indexKeyWidth returns the width of all the columns forming the index key.
+func indexKeyWidth(columns ...[]byte) (width int) {
+	for _, column := range columns {
+		width += len(column)
+	}
+	return
+}
+
+// populateTableIndexKey populates the key passed in with the
+// order encoded values forming the index key.
+func populateTableIndexKey(key []byte, tableID, indexID uint32, columns ...[]byte) []byte {
+	key = append(key, TableDataPrefix...)
+	key = encoding.EncodeUvarint(key, uint64(tableID))
+	key = encoding.EncodeUvarint(key, uint64(indexID))
+	for _, column := range columns {
+		key = append(key, column...)
```

Comment:
You need to use `EncodeBytes` on each of the columns. Just appending them as you've done is incorrect. Consider 2 rows, one with the columns `"a", "ab"` and the other with the columns `"aa", "a"`. Just appending the columns together you get `"aab"` for one key and `"aaa"` for other key and then these sort incorrectly. Using `EncodeBytes` we would get the keys `"a\x00\x01ab\x00\x01"` and `"aa\x00\x01a\x00\x01"` which sort correctly.

The encoding routines have tests for all of this, but you need to add some simple tests for the ordering of the index and data keys.